### PR TITLE
3 - Make more use of `let_chains`

### PR DESCRIPTION
**Pull request by c410-f3r**
Continuation of #94376.

cc #53667

### CODE REVIEW

File: compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs [hunk 1]
```diff
@@ -1086,21 +1086,17 @@ fn get_mut_span_in_struct_field<'tcx>(
...
-                {
-                    return Some(lifetime.span.between(ty.span));
-                }
-            }
-        }
+    if let ty::Ref(_, ty, _) = ty.kind()
+        && let ty::Adt(def, _) = ty.kind()
+        && let field = def.all_fields().nth(field.index())?
+        // Use the HIR types to construct the diagnostic message.
+        && let node = tcx.hir().find_by_def_id(field.did.as_local()?)?
```
Comment (nagisa) [inline compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:1093]: This assignment could be inlined below, right?

Comment (rqytda) [reply compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:1093]: Yes, the same way the old

```rust
let node = tcx.hir().find_by_def_id(field.did.as_local()?)?;
// Now we're dealing with the actual struct that we're going to suggest a change to,
// we can expect a field that is an immutable reference to a type.
if let hir::Node::Field(field) = node {
```

could be inlined to

```rust
// Now we're dealing with the actual struct that we're going to suggest a change to,
// we can expect a field that is an immutable reference to a type.
if let hir::Node::Field(field) = tcx.hir().find_by_def_id(field.did.as_local()?)? {
```

I am trying to transpose everything verbatim

Comment (c410-f3r) [reply compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:1093]: Yes, the same way the old

```rust
let node = tcx.hir().find_by_def_id(field.did.as_local()?)?;
// Now we're dealing with the actual struct that we're going to suggest a change to,
// we can expect a field that is an immutable reference to a type.
if let hir::Node::Field(field) = node {
```

could be inlined to

```rust
// Now we're dealing with the actual struct that we're going to suggest a change to,
// we can expect a field that is an immutable reference to a type.
if let hir::Node::Field(field) = tcx.hir().find_by_def_id(field.did.as_local()?)? {
```

I am trying to transpose everything verbatim

Comment (c410-f3r) [reply compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs:1093]: If you are curious, assembly output is the same for both optimized and un-optimized

**Pull request merged**